src/plugins/discover/public/application/main/services/load_saved_search.ts

```diff
@@ -103,6 +120,10 @@ export const loadSavedSearch = async (
   // Update all other services and state containers by the next saved search
   updateBySavedSearch(nextSavedSearch, deps);
 
+  if (!appState && shouldUpdateWithGlobalFilters) {
+    savedSearchContainer.updateWithFilterManagerFilters({ savedSearch: nextSavedSearch });
+  }
```

Comment:
We could consider to align the usage of this function like `savedSearchContainer.update` is done on L106. just for the case that this function would be extended updating anything else than search source, because than it would be a necessary change.


```suggestion
  if (!appState && shouldUpdateWithGlobalFilters) {
    nextSavedSearch = savedSearchContainer.updateWithFilterManagerFilters({ savedSearch: nextSavedSearch });
  }
```


just FYI, because it's not obvious, but also 
```suggestion
if (!appState && shouldUpdateWithGlobalFilters) {
    nextSavedSearch = savedSearchContainer.update({ useFilterAndQueryServices: true });
  }
```
would do the job here, however it's not obvious and this function in theory does much more, so having a dedicated simple function like this makes sense